### ISSUE TITLE
namespace error with newer fallout 4 scriptnames in starfield

### ISSUE THREAD

**Issue by RedxYeti** [Orvid/Caprica#24]
I'm trying to compile the script Fragments\TopicInfos\tif_ffclinicr02_001c2c28.psc for starfield

the scriptname is
ScriptName Fragments:TopicInfos:TIF_FFClinicR02_001C2C28 Extends TopicInfo Const hidden

--help doesnt have any info on changing the namespace

not sure how to compile these scripts with the newer fallout 4 style scriptnames

**Comment by nikitalita** [comment]
> edit: not sure how possible this is but if no namespace is provided, it would be great if it checked the scriptname for : and automatically set the namespace if found or have an option to create the name space it expected instead of just erroring out

Agreed; after struggling with PCompiler to test out it's default behavior for this (it is incredibly unintuitive), I've decided it would be best to give the option to just recreate the expected namespace directory structure. I'll add this soon.

**Comment by RedxYeti** [comment]
thank you for the option and I finally figured out what I was doing incorrectly.

I didnt realize -o was optimize

changing it to --output fixed my issue 🤦 

I knew the issue was on my end because I got my dir to compile correctly once and then couldn't do it again, but I made this issue out of confusion.

thanks for all your work on this compiler!

**Comment by ForsakenShell** [comment]
Just to add clarity to this:

You need to set the working directory to the root "scripts" directory and compile the scripts with the relative namespace path from that.

eg:
F:\MyMods\ProjectX\Scripts\MyNamespace\ChildNamespace\TheFile.psc

The working directory needs to be:
"F:\MyMods\ProjectX\Scripts"

The scripts must be compiled with the relative path from there:
"MyNamespace\ChildNamespace\TheFile.psc"
